### ISSUE TITLE
Why does reduce.indexed return Record?

### ISSUE THREAD

**Issue by agriffis** [remeda/remeda#153]
I'm confused by this:

https://github.com/remeda/remeda/blob/e961f253d20a9561f7e5ea9bc485c8371377f815/src/reduce.ts#L58-L71

Shouldn't the return type be K?

**Comment by TkDodo** [comment]
you are right, the types do not conform to what happens at runtime. `K` seems to be the correct type - just like for the normal `reduce`. Would you like to quickly PR that maybe?
